code/modules/clothing/spacesuits/rig_modules/rig_modules.dm

```diff
@@ -0,0 +1,90 @@
+	name = "Rig module"
+	desc = "A module to be installed onto a rigsuit."
+	icon = 'icons/obj/module.dmi'
+	icon_state = "std_mod"
+	var/mob/living/wearer
+	var/obj/item/clothing/suit/space/rig/rig
+
+/obj/item/rig_module/proc/examine_addition(mob/user)
+
+/obj/item/rig_module/proc/activate(var/mob/user,var/obj/item/clothing/suit/space/rig/R)
+	wearer = user
+	rig = R
+
+/obj/item/rig_module/proc/deactivate(var/mob/user, var/obj/item/clothing/suit/space/rig/R)
+	wearer = null
+	rig = null
+
+/obj/item/rig_module/speed_boost
+	name = "Rig speed module"
+	desc = "Self-lubricating joints allow for ease of movement when walking in a rigsuit."
+
+/obj/item/rig_module/speed_boost/activate(var/mob/user,var/obj/item/clothing/suit/space/rig/R)
+	if(R.cell.use(500))
+		to_chat(user, "<span class = 'binaryradio'>Speed module engaged.</span>")
+		R.slowdown = max(1, slowdown/2)
+
+/obj/item/rig_module/speed_boost/deactivate(var/mob/user, var/obj/item/clothing/suit/space/rig/R)
+	R.slowdown = initial(R.slowdown)
+
+/obj/item/rig_module/health_readout
+	name = "articulated spine"
+	desc = "Lets passers by read your health from a distance"
+
+/obj/item/rig_module/health_readout/examine_addition(mob/user)
+	if(wearer)
+		to_chat(user, "<span class = 'notice'>The embedded health readout reads: [(wearer.health/wearer.maxHealth)*100]%</span>")
+
+/obj/item/rig_module/tank_refiller
+	name = "tank pressurizer"
+	desc = "When in atmosphere, syphons from the air to refill the tank connected to your internals."
+	var/gas_id
+
+/obj/item/rig_module/tank_refiller/activate(var/mob/user, var/obj/item/clothing/suit/space/rig/R)
+	..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	if(H.internal)
+		var/datum/organ/internal/lungs/L = H.internal_organs_by_name["lungs"]
+		if(L)
+			var/datum/lung_gas/metabolizable/M = locate() in L.gasses
+			gas_id = M.id
+		processing_objects.Add(src)
+		to_chat(user, "<span class = 'binaryradio'>Internals pressurizer online. Syphoning [gas_id] from environment to [H.internal]</span>")
+	else
+		to_chat(user, "<span class = 'binaryradio'>Internals pressurizer failed to find internals. Aborting.</span>")
+
+/obj/item/rig_module/tank_refiller/deactivate(var/mob/user, var/obj/item/clothing/suit/space/rig/R)
+	..()
+	if(processing_objects.Find(src))
+		processing_objects.Remove(src)
+
+/obj/item/rig_module/tank_refiller/process()
+	if(!wearer || !ishuman(wearer))
+		processing_objects.Remove(src)
+
+	var/mob/living/carbon/human/H = wearer
+	if(!H.internal)
+		to_chat(H, "<span class = 'binaryradio'>Internals pressurizer failed to find internals. Aborting.</span>")
+		processing_objects.Remove(src)
+	else
+		var/datum/gas_mixture/M = H.loc.return_air()
+		if(M.gas[gas_id] && rig.cell.use(50))
+			var/datum/gas_mixture/internals = H.internal.air_contents
+
+			var/pressure_delta = 10*ONE_ATMOSPHERE - internals.pressure
+			var/transfer_moles = pressure_delta * internals.volume / (M.temperature * R_IDEAL_GAS_EQUATION)
```

Comment:
This method results in the performance *not* depending on the pressure of your surroundings, but still depending on the concentration of the desired gas in the surroundings
This seems weird but if it's intended, OK

Reply:
It is intended, it's so the tank doesn't explode. 10*ONE_ATMOSPHERE is the overall 'safe point' that is used in places like canisters.

Reply:
That's not what I mean. This sucks up all available air in a zone, regardless of how thin it is, *but* chokes if the desired gas is a low proportion of all gas because it calculates this based on the whole mixture but only takes the desired gas.
I'd really recommend using `remove_volume()` to get the available mixture and then preventing the explosion in the same step as taking the desired gas from the available by just not moving more gas than the tank can take into the tank

Reply:
This is still here